src/Model/lbsProcessor.py

```diff
@@ -251,28 +251,32 @@ def process_underload_message(self, msg):
 
     def compute_cmf_underloads(self, l_ave, pmf_type=0):
         """Compute CMF of underloads given an average load
+           tyoe 0: improved Gossip approac=h
+           type 1: NS variant based on sender load
         """
 
         # Initialize CMF
+        sum_p = 0
         cmf = []
+        p_fac = 1
 
+        # Retrieve known underloads
+        loads = self.known_underloads.values()
+        
         # Distinguish between different PMF types
         if not pmf_type:
             # Determine whether one underloaded is actually overloaded
-            loads = self.known_underloads.values()
             l_max = max(loads)
-            factor = l_max if l_max > l_ave else l_ave
-                
-            # Initialize CMF value
-            sum_p = 0.
-            
-            # Iterate over all loads known to be underloaded
-            for l in loads:
-                # Update CMF
-                sum_p += 1. - factor * l
+            p_fac /= (l_max if l_max > l_ave else 1 / l_ave)
```

Comment:
This looks suspicious to me, one option being `l_max` and the other being `1 / l_ave`.

Reply:
Indeed !